src/App.tsx

```diff
@@ -3,29 +3,107 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { GetTodo } from './api/api';
 
 interface State {
+  query: string;
+  showOnly: string;
+  todos: Todo[] | [];
   selectedUserId: number;
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
+    query: '',
+    showOnly: '',
+    todos: [],
     selectedUserId: 0,
   };
 
+  componentDidMount() {
+    GetTodo()
+      .then((response) => {
+        this.setState({ todos: response });
+      });
+  }
+
+  handleChange = (userId: number) => {
+    this.setState({ selectedUserId: userId });
+  };
+
+  handleInput = (event: React.ChangeEvent<HTMLInputElement> |
+  React.ChangeEvent<HTMLSelectElement>) => {
+    const { value, name } = event.target;
+
+    this.setState({
+      [name]: value,
+    } as unknown as Pick<State, keyof State>);
+  };
+
+  clearUser = () => {
+    this.setState({
+      selectedUserId: 0,
+    });
+  };
+
+  sortBy = (filterby: string) => {
+    const filteredTodos = [...this.state.todos].filter(todo => (
+      (todo.title.toLocaleLowerCase().includes(this.state.query.toLocaleLowerCase()))));
```

Comment:
for this.state.query.toLocaleLowerCase() better make a separate variable not to calculate it on every iteration